File: python/program/Dog.py

from Pet import Pet

class Dog(Pet):
    def __init__(self, species="", name="", owner="", vaccinated=False, breed="", weight=0.0):
        super().__init__(species, name, owner, vaccinated)
        self.__breed = breed
        self.__weight = weight
    
    def set_breed(self, breed):
        self.__breed = breed
    
    def set_weight(self, weight):
        self.__weight = weight
    
    def get_breed(self):
        return self.__breed
    
    def get_weight(self):
        return self.__weight

File: java/program/Dog.java

public class Dog extends Pet {
    private String breed;
    private float weight;
    
    public Dog() {
    }
    
    public Dog(String species, String name, String owner, boolean vaccinated, String breed, float weight) {
        super(species, name, owner, vaccinated);
        this.breed = breed;
        this.weight = weight;
    }
    
    public void setBreed(String breed) {
        this.breed = breed;
    }
    
    public void setWeight(float weight) {
        this.weight = weight;
    }
    
    public String getBreed() {
        return this.breed;
    }
    
    public float getWeight() {
        return this.weight;
    }
}